### ISSUE TITLE
Considering GPLv3 Licensing 

### ISSUE THREAD

**Issue by edo0** [qarmin/czkawka#347]
I personally really like Czkawka, and have used it a great deal to clean up my storage.

I was wondering why not using the GPLv3 (or other versions of the GPL license), to make sure your code is protected from unauthorized integration by third parties into proprietary software - which the MIT license does not protect against.

If you were instead open to companies using it for commercial software, you may still release it as GPL, and dual license it if approached, so as to enabling the commercial application at your conditions and following own ideas of remuneration. 

I hope I made sense, and yes, it's really mostly a curiosity of mine as to why going with MIT over GPL
Thanks again for you work on Czkawka @qarmin

**Comment by lodenrogue** [comment]
I second this. At the very least the UI can be GPL and the library itself can be AGPL.

**Comment by qarmin** [comment]
In general, I like the GPL license, but I have nothing against MIT either.
Due to the fact that I am not very familiar with all the license terms, I decided to use the MIT license, because all the libraries I use are under this license.
Additionally, from what I remember, MIT allows to change the license quite easily.

**Comment by edo0** [comment]
Yes, that is absolutely correct! MIT allows to change licence quite easily, since it is a permissive license. With that comes also that anyone can do so, which can clearly be a bit of a double-edged sword, as contrasted with the GPL which allows only licensing under the same license (being a non-permissive, strong copyleft license).

The latter is known to be more resilient to potential commercial overtake, and is ideal if you plan to keep the software and its derivatives free and open source software, since any derivative work, paid or free, would anyway have to release its source code publicly. 
It's also true that if you wanted to [relicense](https://softwareengineering.stackexchange.com/questions/148146/open-source-code-with-no-license-can-i-fork-it) or [dual license](https://opensource.stackexchange.com/questions/4092/how-to-accept-contributions-for-agpl-proprietary-dual-licensed-projects) a GPL project, you need to have rights to all of its code, which means you will have to accept third party contributions with a CLA (Contributor License Agreement) transferring rights to you, which can be [integrated](https://github.com/cla-assistant/cla-assistant) in the template of the pull request.

>In order to offer both licenses, you would need to be and stay the primary or sole copyright holder. You would effectively need a CLA where contributors grant you enough rights to relicense using both licenses or assign the copyrights to you. 

An example of software that does this is GPL-3.0 License IM app Signal ([here is its CLA](https://signal.org/cla/))

Again, if you don't have commercial interests, a plain GPL would be the simplest way. But as you can see, there is also the possibility I explained above, allowing protection of the software, while granting to yourself only the rights to dual licensing (or relicensing) for commercial purposes.

**Comment by qarmin** [comment]
Closing as wontfix
The MIT license is sufficient for this project - the GTK and CLI versions use this license.
The MIT license is also used by the Krokiet source code, but the entire application is licensed under GPL 3 (because of the Slint license).
